alacritty/src/display/window.rs

```diff
@@ -163,14 +163,19 @@ impl Window {
             window_builder = window_builder.with_embed_parent_window(parent_window_id);
         }
 
+        #[cfg(target_os = "macos")]
+        let fullscreen = if tabbing_id.is_some() { None } else { config.window.fullscreen() };
```

Comment:
You're just disabling fullscreen when tabbing is enabled, but that can be easily done by the user themselves if that's actually what they want.

I don't know how macOS works here exactly, but I would want the fullscreen window to get the tabs. I'm not actually sure what happens when a window with multiple tabs is made fullscreen, but I'd assume it doesn't just get split into multiple windows?

Also if we wanted to prohibit having these two things set together, it should be a configuration warning/error at least. We shouldn't just make these changes and not report back to the user that their selected configuration is invalid.

Reply:
If you look at the video in the issue you will see that when he create a tab that make another window into fullscreen mode so we should disable tab when start mode fullscreen is set or make it a config

Also when making alacritty fullscreen without startup mode tabs work perfectly fine

And when startup mode is set exiting alacritty leave buggy window in fullscreen while alacritty is exited